### PR TITLE
Add Model Revision Support

### PR DESCRIPTION
**Pull request by ghost**
This PR adds an additional attribute to the LLM engine, the `revision` options download specify the commit version for the model for consistency and reliability.

For API_Server usage
```shell
python3 -m vllm.entrypoints.api_server --model facebook/opt-125m --revision 507a3991d874042a92e7581eb6e7cc7074b0c77e
```

For LLM Engine usage
```python
from vllm import LLM, SamplingParams

llm = LLM(model="facebook/opt-125m", revision="507a3991d874042a92e7581eb6e7cc7074b0c77e")
sampling_params = SamplingParams(temperature=0.8, top_p=0.95)
outputs = llm.generate(["hello, my name is "], sampling_params)

### CODE REVIEW

**Comment by JasLoh564** [comment]
Hi @WoosukKwon, 

Could you review this PR 

Thanks.

File: vllm/engine/arg_utils.py [hunk 1]
```diff
@@ -48,6 +49,12 @@ def add_cli_args(
             type=str,
             default=EngineArgs.tokenizer,
             help='name or path of the huggingface tokenizer to use')
+        parser.add_argument(
+            '--revision',
+            type=str,
+            default=None,
```
Comment (zhuohan123) [inline vllm/engine/arg_utils.py:55]: Can you make the default value consistent across the code base? It's `None` here but `"main"` everywhere else.

File: vllm/engine/arg_utils.py [hunk 2]
```diff
@@ -48,6 +49,12 @@ def add_cli_args(
...
+            help='The specific model version to use. It can be a branch '
+            'name, a tag name, or a commit id.')
```
Comment (zhuohan123) [inline vllm/engine/arg_utils.py:57]: Nit: In the help message, can you add a description of the default behavior? Bascially add "By default, we use the main branch."

**Review by zhuohan123 — APPROVED** [review]
Thank you for your contribution!  In general LGTM! Left a small comment on the default value. We can merge this branch after that is fixed. Also, please format your script with `format.sh`

**Pull request merged**